swift/example_code/s3/basics/Tests/basics-tests/S3DemoCleanup.swift

```diff
@@ -0,0 +1,225 @@
+/*
+   A class to record the Amazon S3 buckets and files created by a code example,
+   so that they can easily be cleaned up after the example is finished.
+
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+import Foundation
+import AWSS3
+import ClientRuntime
+import AWSClientRuntime
+import SwiftUtilities
+
+/// Errors thrown by the `S3DemoCleanup` class.
+enum S3DemoCleanupError: Error {
+    case duplicate          /// Attempted to create an item that already exists.
+    case notFound           /// Attempted to access an item that doesn't exist.
+}
+
+/// A structure identifying a file given its bucket name and the
+/// file's name within the bucket.
+public struct S3DemoFileInfo: Equatable {
+    /// The name of the bucket containing the file.
+    var bucket: String
+    /// The file's name.
+    var file: String
+    /// The contents of the file.
+    var contents: Data?
+
+    init(bucket: String, file: String, contents: Data? = nil) {
+        self.bucket = bucket
+        self.file = file
+        self.contents = contents
+    }
+
+    /// Determine if the `S3DemoFileInfo` matches the given bucket
+    /// name and file name.
+    ///
+    /// - Parameters:
+    ///   - bucket: Bucket name to compare to.
+    ///   - file: File name to compare to.
+    func matches(bucket: String, file: String) -> Bool {
+        return bucket == self.bucket && file == self.file
+    }
+}
+
+public class S3DemoCleanup {
+    /// Returns a single shared instance of `S3DemoCleanup` that should
+    /// always be used to access the cleanup manager.
```

Comment:
Is "cleanup manager" the same thing as the "test cleanup processor" mentioned in line 59? If so, recommend using the same term for consistency (whichever is more accurate).